pkgs/by-name/se/seabios/package.nix

```
@@ -29,31 +58,42 @@ stdenv.mkDerivation (finalAttrs: {
     "EXTRAVERSION=\"-nixpkgs\""
   ];
 
-  hardeningDisable = [ "pic" "stackprotector" "fortify" ];
+  hardeningDisable = [
+    "fortify"
+    "pic"
+    "stackprotector"
+  ];
 
-  postConfigure = let
-    config = writeText "config.txt" (lib.generators.toKeyValue { } {
-      # SeaBIOS with CSM (Compatible Support Module) support; learn more at
-      # https://www.electronicshub.org/what-is-csm-bios/
-      "CONFIG_CSM" = "y";
-      "CONFIG_PERMIT_UNALIGNED_PCIROM" = "y";
-      "CONFIG_QEMU_HARDWARE" = "y";
-    });
-  in ''
-    cp ${config} .config
-    make olddefconfig
-  '';
+  postConfigure =
+    let
+      config = writeText "config.txt" (
+        lib.generators.toKeyValue { } {
+          "${configuration-string}" = "y";
+          "CONFIG_PERMIT_UNALIGNED_PCIROM" = "y";
+          "CONFIG_QEMU_HARDWARE" = "y";
+        }
+      );
+    in
+    ''
+      cp ${config} .config
+      make olddefconfig
+    '';
 
   installPhase = ''
     runHook preInstall
 
-    mkdir -pv $doc/share/doc/seabios-${finalAttrs.version}/
-    cp -v docs/* $doc/share/doc/seabios-${finalAttrs.version}/
-    install -Dm644 out/Csm16.bin -t $out/share/seabios/
+    mkdir -pv ''${!outputDoc}/share/doc/seabios-${finalAttrs.version}/
+    cp -v docs/* ''${!outputDoc}/share/doc/seabios-${finalAttrs.version}/
+    install -Dm644 out/${biosfile} -t $out/share/seabios/
 
     runHook postInstall
   '';
 
+  passthru = {
+    build-type = ___build-type;
+    firmware-location = "${finalAttrs.finalPackage}/share/seabios/${biosfile}";
```

```suggestion
    firmware = "${finalAttrs.finalPackage}/share/seabios/${biosfile}";
```
For parity with OVMF.

---

OK!